qt/MainWindow.cpp

```diff
@@ -661,6 +661,21 @@ void SC_MainWindow::startNewImport( const QString& region, const QString& realm,
   }
   simProgress = 0;
   import_sim  = initSim();
+
+  #ifndef SC_NO_NETWORKING
+  // Try to import the API key from the GUI options
+  if (import_sim->apikey == "") {
+    QSettings settings;
+    std::string apikey = settings.value("options/api_client_id").toString().toStdString() +
```

Comment:
I think you should get the value not from the settings object, but from optionsTab->api_client_id->currentText() and optionsTab->api_client_secret->currentText(). Otherwise this will not pick up the value a user just entered into the QLineEdit control. Settings are only saved when a simulation starts / gui is closed.

To check if a std::string is empty you can just call its empty() member function instead of comparing to "".

Reply:
Ok I've resolved this one :)

Getting into projects is always quite a shock, but I'll kick in (I hope :D )